src/app/feedback/viewfeedback.component.ts

```diff
@@ -0,0 +1,149 @@
+          return feedback.doc;
+        }).filter(fback  => {
+          return fback['_id'].indexOf('_design') !== 0;
+        });
+      }, (error) => this.message = 'There is a problem of getting data.');
+  }
+
+  deleteClick(feedback) {
+    this.deleteDialog = this.dialog.open(DialogsDeleteComponent, {
+      data: {
+        okClick: this.deleteFeedback(feedback),
+        type: 'feedback',
+        displayName: feedback.type
+      }
+    });
+    // Reset the message when the dialog closes
+    this.deleteDialog.afterClosed().debug('Closing dialog').subscribe(() => {
+      this.message = '';
+    });
+  }
+
+  deleteFeedback(feedback) {
+    // Return a function with feedback on its scope so it can be called from the dialog
+    return () => {
+      const { _id: feedbackId, _rev: feedbackRev } = feedback;
+      this.couchService.delete(this.dbName + '/' + feedbackId + '?rev=' + feedbackRev)
+        .subscribe((data) => {
+          // It's safer to remove the item from the array based on its id than to splice based on the index
+          this.feedback.data = this.feedback.data.filter((fback: any) => data.id !== fback._id);
+          this.deleteDialog.close();
+        }, (error) => this.deleteDialog.componentInstance.message = 'There is a problem deleting this feedback.');
+    };
+  }
+
+  back() {
+    this.location.back();
+  }
+
+  reply(feedback) {
+    const title = 'Reply';
+    const type = 'feedback';
+    const fields =
+      [
+        { 'label': 'To', 'type': 'textbox', 'name': 'replyTo', 'readonly': true, 'value': feedback.owner , 'required': false },
+        { 'label': 'Message', 'type': 'textarea', 'name': 'message', 'placeholder': 'Leave a comment', 'required': true }
+      ];
+    const validation = {
+      id: [ feedback._id ],
+      replyTo: [ feedback.owner ],
+      message: [ '', Validators.required ]
+    };
+    this.dialogsFormService
+      .confirm(title, type, fields, validation, '')
+      .debug('Dialog confirm')
+      .subscribe((res) => {
+        if (res !== undefined) {
+          this.onSubmit(res);
+        }
+      });
+  }
+
+  onSubmit(feedback) {
+    const messages = [];
+    this.couchService.get(this.dbName + '/' + feedback.id)
+      .subscribe((data) => {
+        data.messages.push({ 'message': feedback.message, 'time': Date.now(), 'user': this.userService.get().name });
+        this.couchService.put(this.dbName + '/' + data._id, {  ...data })
+        .subscribe(() => {
+          this.message = 'Reply success.';
+        }, (error) => this.message = '');
+      }, (error) => this.message = 'There is a problem of getting data.');
+  }
+
+  openFeedback(feedback: any) {
+    feedback.selected = feedback.selected ? false : true;
+    const msg = feedback.messages.map(message => {
+      return message;
+    });
+    this.dialogsMessage = this.dialog.open(DialogsMessageComponent, {
+      data: {
+        type: 'FeedBack',
+        Msg: msg
```

Comment:
can we have keys of dialogsMessage changed such that it will have same formatting.